src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -933,7 +933,53 @@ public static bool IsPointerType(this TypeSymbol type)
 
         internal static int FixedBufferElementSizeInBytes(this TypeSymbol type)
         {
-            return type.SpecialType.FixedBufferElementSizeInBytes();
+            var specialType = type.SpecialType;
+
+            if (specialType != SpecialType.None)
+            {
+                return specialType.FixedBufferElementSizeInBytes();
+            }
+
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                // TODO: Add Diagnostic
+                return 0;
+            }
+
+            // TODO: Do we need to handle explicit sizing of structs (those that are
+            // marked with `StructLayout(LayoutKind.Sequential, Pack = <blah>, Size = <blah>)`
+
+            var members = type.GetMembers();
+
+            int sizeInBytes = 0;
+
+            foreach (var member in members)
```

Comment:
We shouldn't be in the business of calculating struct size because only runtime decides what struct layout should be. All struct types used for interop should have an explicit layout, including size. If we really want to support fixed size buffers for structures, I think we should support them only for structures with explicit size.